source/normalize-arguments.js

```diff
@@ -51,6 +51,16 @@ module.exports = (url, options, defaults) => {
 		...options
 	};
 
+	const {baseUrl} = options;
+	Object.defineProperty(options, 'baseUrl', {
+		set: _ => {
+			throw new Error('Failed to set baseUrl. Options are normalized already.');
+		},
+		get: () => {
+			return baseUrl;
+		}
+	});
+
```

Comment:
The same result we can achieve using:

```js
Object.defineProperty(options, 'baseUrl', {
	writable: false
});
```

But the error will just say it's not writable, no other details.